### DISCUSSION TITLE
how to get the  JSON Schema's defaultValue in [x-component-props] when I do submit?

### DISCUSSION THREAD

**Discussion by fongfai** [alibaba/formily#1915] [Q&A 2.x] [answered]
I got empty value when I click Submit button,  codesandbox: https://codesandbox.io/s/peaceful-violet-gfyh3?file=/App.tsx

if cant get the shema's default value,  is there has any method to extract JSON Schema‘s defaultValue？

**Comment by janryWang** [comment, marked as answer]
please use schema.default property

**Reply by fongfai** [reply]
thanks!

**Reply by fongfai** [reply]
> please use schema.default property

look at this;  I just set a default(this represent the  initialValue in this document: https://react.formilyjs.org/zh-CN/api/shared/schema ), why  onFormValuesChange be triggered? for me, I think there is just should the onFormInitialValuesChange,  is this a bug?  
https://codesandbox.io/s/formily-default-demo-st15f?file=/App.tsx:243-261

**Reply by janryWang** [reply]
This is the design because the default value will be merged into the value
